packages/truffle-box/lib/utils/unbox.js

```diff
@@ -53,20 +49,20 @@ function verifyURL(url) {
 }
 
 function fetchRepository(url, dir) {
-  return new Promise(function(accept, reject) {
+  return new Promise((accept, reject) => {
     // Download the package from github.
     ghdownload(url, dir)
-      .on("err", function(err) {
+      .on("err", err => {
         reject(err);
       })
-      .on("end", function() {
+      .on("end", () => {
         accept();
       });
   });
 }
 
-function prepareToCopyFiles(tempDir, boxConfig) {
-  const needingRemoval = boxConfig.ignore || [];
+function prepareToCopyFiles(tempDir, { ignore }) {
+  const needingRemoval = ignore;
```

Comment:
Do we need to keep this behavior of having the default for `needingRemoval` set to an empty array?

Reply:
Looks like `ignore` is either an empty array or an actual list of items to be ignored during unboxing (depending on the given truffle box being unpacked).